central/compliance/datastore/internal/store/postgres/store.go

```diff
@@ -87,7 +101,48 @@ func (s *storeImpl) GetConfig(ctx context.Context, id string) (*storage.Complian
 	return s.config.Get(ctx, id)
 }
 
-func (s *storeImpl) getResultsFromMetadata(ctx context.Context, metadata *storage.ComplianceRunMetadata, flags types.GetFlags) (*storage.ComplianceRunResults, error) {
+func getRunResultsCache() lru.Cache[string, *storage.ComplianceRunResults] {
+	onceforRunResults.Do(func() {
+		runResultsCache = lru.NewExpirableLRU[string, *storage.ComplianceRunResults](
+			runResultsCacheSize,
+			nil,
+			runResultsCacheRetention,
+		)
+	})
+	return runResultsCache
+}
+
+func (s *storeImpl) getResultsFromMetadata(
+	ctx context.Context,
+	metadata *storage.ComplianceRunMetadata,
+	flags types.GetFlags,
+) (*storage.ComplianceRunResults, error) {
+	scopeChecker := sac.GlobalAccessScopeChecker(ctx).AccessMode(storage.Access_READ_ACCESS).Resource(resources.Compliance)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.View(resources.Compliance))
```

Comment:
One consideration: having flags in a key is needed. But I'm not sure about the scope. We are introducing additional checks which are (in my opinion) redundant. Users should not retrieve Results Metadata if they don't have sufficient rights. I would expect that scope is already checked before this function is called. If you take a look at `getDomain` - we are not checking the scope there.

If you still think it's required and safer to be used. Then we should not change the existing behaviour of `getResultsFromMetadata` if `scopeChecker.EffectiveAccessScope` returns an error.

What I want to say is. If `scopeChecker.EffectiveAccessScope`, we should not cache that result but still return what we get from `getResultsFromMetadataFromStore`.

Reply:
The flow was altered to silently ignore `scopeChecker.EffectiveAccessScope` failures.
I think chances are high that the store lookup will face the same error further down the flow. But in case the error was transient, it increases the chances of getting the results anyway.